### PR TITLE
`GraphEdit`: Scrollwheel (w/o ctrl) to zoom, mouse warping when panning

### PR DESCRIPTION
**Pull request by AaronRecord**
I changed zooming to just use the scrollwheel to make it consistent with the 2d editor. You can still scroll with:
<kbd>Ctrl + Scroll = Vertical Pan</kbd>
<kbd>Ctrl + Shift + Scroll = Horizontal Pan</kbd>

Also converted some macros to constants.

### CODE REVIEW

File: scene/gui/graph_edit.cpp
Comment (EricEzaM) [inline scene/gui/graph_edit.cpp:49]: Can't comment on this, a core dev who decides code style stuff might have to give their thoughts.

File: scene/gui/graph_edit.cpp
Comment (EricEzaM) [inline scene/gui/graph_edit.cpp:1336]: I am fine with removing the Control modifier requirement (and am of the opinion it is the right way to go), but some people won't be - there is no winner to this battle.

What I don't understand though is removing all the vertical and horizontal scrolling functionality? That is legitimately useful, why remove it? In this case I think the Blender node editor serves as a good example:
```
Scroll = Zoom
Ctrl + Scroll = Horizontal Pan
Shift + Scroll = Vertical Pan
```

File: scene/gui/graph_edit.cpp
Comment (Calinou) [reply scene/gui/graph_edit.cpp:49]: `const` (or better, `constexpr`) should be preferred over symbolic defines in modern C++.

File: scene/gui/graph_edit.cpp
Comment (LightningAA) [reply scene/gui/graph_edit.cpp:1336]: I didn't know Blender did that, that's an interesting idea, I'll do that

File: scene/gui/graph_edit.cpp
Comment (LightningAA) [reply scene/gui/graph_edit.cpp:49]: I would have done `constexpr` but it seems it only appears in thirdparty code from what I can see, should I use `constexpr`?

Comment (KoBeWi) [reply scene/gui/graph_edit.cpp:49]: There are some plans to change some macros to `constexpr` (see #45263 for example), so it's ok to use them.

File: scene/gui/graph_edit.cpp
Comment (LightningAA) [reply scene/gui/graph_edit.cpp]: I made it
```
Ctrl + Scroll = Vertical Pan
Shift + Scroll = Horizontal Pan
```
(See https://chat.godotengine.org/channel/editor?msg=9t53ykBiAAJtzpA2s)

File: scene/gui/graph_edit.cpp
Comment (LightningAA) [inline scene/gui/graph_edit.cpp:43]: Wasn't sure if these should be float, double, or real_t

**Comment by LightningAA** [comment]
> > You can still scroll with:
> > Ctrl + Scroll = Vertical Pan
> > Shift + Scroll = Horizontal Pan
> 
> Maybe horizontal panning should be Shift + Ctrl + Scroll, to keep with the semantic meaning of the buttons? Like Ctrl + Scroll is for panning, and you add Shift to shift the panning mode.
> 
> Edit: It should also help simplify the code, as you would use Shift to only switch between `h_scroll` and `v_scroll`, with the rest staying more or less the same.

Ctrl + Scroll was what we'd discussed on rocket chat, but I  like that better.

**Comment by pycbouh** [comment]
> Ctrl + Scroll was what we'd discussed on rocket chat, but I like that better.

Huh? I'm not arguing against Ctrl + Scroll. I'm arguing that for horizontal panning instead of Shift + Scroll it should be Shift + Ctrl + Scroll, for the reasons explained.

**Comment by LightningAA** [comment]
Oh yeah, I understood what you meant, my response was just confusing.

**Comment by LightningAA** [comment]
@pycbouh Alright, I changed it be how you suggested.

**Comment by pycbouh** [comment]
Code looks much clearer now, and warping should be a good feature. I have no idea about the benefits of `constexpr`s though. And this also needs a rebase before I can approve it.

**Comment by LightningAA** [comment]
Rebased, see https://github.com/godotengine/godot/pull/47986#discussion_r629411618 for the discussion on `constexpr`.

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**
